full-service/src/test_utils.rs

```diff
@@ -658,21 +669,36 @@ pub fn get_resolver_factory(
 pub fn setup_wallet_service(
     ledger_db: LedgerDB,
     logger: Logger,
+) -> WalletService<MockBlockchainConnection<LedgerDB>, MockFogPubkeyResolver> {
+    setup_wallet_service_impl(ledger_db, logger, false)
+}
+
+pub fn setup_wallet_service_offline(
+    ledger_db: LedgerDB,
+    logger: Logger,
+) -> WalletService<MockBlockchainConnection<LedgerDB>, MockFogPubkeyResolver> {
+    setup_wallet_service_impl(ledger_db, logger, true)
+}
+
+fn setup_wallet_service_impl(
+    ledger_db: LedgerDB,
+    logger: Logger,
+    offline: bool,
 ) -> WalletService<MockBlockchainConnection<LedgerDB>, MockFogPubkeyResolver> {
     let mut rng: StdRng = SeedableRng::from_seed([20u8; 32]);
 
     let db_test_context = WalletDbTestContext::default();
     let wallet_db = db_test_context.get_db_instance(logger.clone());
     let (peer_manager, network_state) =
-        setup_peer_manager_and_network_state(ledger_db.clone(), logger.clone());
+        setup_peer_manager_and_network_state(ledger_db.clone(), logger.clone(), offline);
 
     WalletService::new(
         wallet_db,
         ledger_db,
         peer_manager,
         network_state,
         get_resolver_factory(&mut rng).unwrap(),
-        false,
+        offline,
```

Comment:
oh, interesting. There was an 'offline' option already, but it didn't actually change anything?